BUILD FAILED (Android)

Hello,

When I run:
`npx expo run:android --device`

I have the following error:
`
› Building app...
Configuration on demand is an incubating feature.
Kotlin does not yet support 22 JDK target, falling back to Kotlin JVM_21 JVM target

> Task :react-settings-plugin:compileKotlin FAILED
Kotlin does not yet support 22 JDK target, falling back to Kotlin JVM_21 JVM target

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':react-settings-plugin:compileKotlin'.
> Inconsistent JVM-target compatibility detected for tasks 'compileJava' (22) and 'compileKotlin' (21).
  
  Consider using JVM Toolchain: https://kotl.in/gradle/jvm/toolchain
  Learn more about JVM-target validation: https://kotl.in/gradle/jvm/target-validation

BUILD FAILED
`

--- a/apps/expo-multi-tv/app.json
+++ b/apps/expo-multi-tv/app.json
@@ -1,6 +1,7 @@
 {
   "expo": {
     "plugins": [
+      "./plugins/withKotlinJvmTarget",
       "@bam.tech/react-native-keyevent-expo-config-plugin",
       [
         "@react-native-tvos/config-tv",
@@ -24,7 +25,11 @@
             "newArchEnabled": false
           },
           "android": {
-            "newArchEnabled": false
+            "newArchEnabled": false,
+            "kotlinVersion": "1.9.24",
+            "compileSdkVersion": 34,
+            "targetSdkVersion": 34,
+            "minSdkVersion": 23
           }
         }
       ],

--- a/apps/expo-multi-tv/plugins/withKotlinJvmTarget.js
+++ b/apps/expo-multi-tv/plugins/withKotlinJvmTarget.js
@@ -1,0 +1,44 @@
+const { withDangerousMod } = require("expo/config-plugins");
+const fs = require("fs");
+const path = require("path");
+
+function withKotlinJvmTarget(config) {
+  return withDangerousMod(config, [
+    "android",
+    async (config) => {
+      const buildGradlePath = path.join(
+        config.modRequest.platformProjectRoot,
+        "react-settings-plugin",
+        "build.gradle.kts"
+      );
+
+      if (fs.existsSync(buildGradlePath)) {
+        let contents = fs.readFileSync(buildGradlePath, "utf-8");
+
+        if (!contents.includes("sourceCompatibility")) {
+          const jvmConfig = `
+java {
+    sourceCompatibility = JavaVersion.VERSION_17
+    targetCompatibility = JavaVersion.VERSION_17
+}
+
+tasks.withType<org.jetbrains.kotlin.gradle.tasks.KotlinCompile> {
+    kotlinOptions {
+        jvmTarget = "17"
+    }
+}
+`;
+          contents = contents.replace(
+            /repositories\s*\{\s*mavenCentral\(\)\s*\}/,
+            (match) => match + "\n" + jvmConfig
+          );
+          fs.writeFileSync(buildGradlePath, contents);
+        }
+      }
+
+      return config;
+    },
+  ]);
+}
+
+module.exports = withKotlinJvmTarget;

